### PR TITLE
DOC: empty: standardize notes about uninitialized values

### LINKED ISSUE #21482: DOC: Warn about empty values in `empty_like`, `masked_all`, `masked_all_like` like `np.ma.empty` does
### Issue with current documentation:

Source:
https://numpy.org/doc/stable/reference/generated/numpy.ma.empty.html

Destinations:
https://numpy.org/doc/stable/reference/generated/numpy.ma.empty_like.html
https://numpy.org/doc/stable/reference/generated/numpy.ma.masked_all.html
https://numpy.org/doc/stable/reference/generated/numpy.ma.masked_all_like.html

### Idea or request for content:

I've recently found out the hard way that this warning in [`np.ma.empty`](https://numpy.org/doc/stable/reference/generated/numpy.ma.empty.html
):
> [empty](https://numpy.org/doc/stable/reference/generated/numpy.empty.html#numpy.empty), unlike [zeros](https://numpy.org/doc/stable/reference/generated/numpy.zeros.html#numpy.zeros), does not set the array values to zero, and may therefore be marginally faster. On the other hand, it requires the user to manually set all the values in the array, **and should be used with caution.**

is missing from the `np.ma` docs of [`empty_like`](https://numpy.org/doc/stable/reference/generated/numpy.ma.empty_like.html
), [`masked_all`](https://numpy.org/doc/stable/reference/generated/numpy.ma.masked_all.html
), [`masked_all_like`](https://numpy.org/doc/stable/reference/generated/numpy.ma.masked_all_like.html).

"Garbage values" are quite rare in python, and it's easy to forget about them.
I never had to use `empty`, but I did find `masked_all` useful for edge cases in masked averaging.
Then one day some change in my code unmasked those empty values, which I'd forgotten about.
This has led to a classic [Heisenbug](https://en.wikipedia.org/wiki/Heisenbug). (in my case, outputs differing between running and debugging)

Future travelers who seek fast and safe behavior of `all_masked` + `zeros`, this seems to do it:
```python
def all_masked_zeros(shape: Union[int, Tuple[int, ...]]) -> np.ma.MaskedArray:
    mat = np.ma.zeros(shape)
    mat[:] = np.ma.masked
    return mat
```

### PR DESCRIPTION
**Pull request by mdhaber**
gh-21482 reported that some functions similar to `np.empty` were missing notes about uninitialized values. This PR standardizes the note about uninititalized values for functions I could find that were similar to `np.empty`. (If I missed any, LMK.)

Closes gh-21482

### CODE REVIEW

File: numpy/ma/extras.py [hunk 1]
```diff
@@ -122,6 +122,15 @@ def masked_all(shape, dtype=float):
     --------
     masked_all_like : Empty masked array modelled on an existing array.
 
+    Notes
+    -----
+    Unliked other masked array creation functions (e.g. `numpy.ma.zeros`,
```
Comment (rossbar) [inline numpy/ma/extras.py:127]:
```suggestion
    Unlike other masked array creation functions (e.g. `numpy.ma.zeros`,
```

File: numpy/ma/extras.py [hunk 2]
```diff
@@ -177,6 +186,15 @@ def masked_all_like(arr):
     --------
     masked_all : Empty masked array with all elements masked.
 
+    Notes
+    -----
+    Unliked other masked array creation functions (e.g. `numpy.ma.zeros_like`,
```
Comment (rossbar) [inline numpy/ma/extras.py:191]:
```suggestion
    Unlike other masked array creation functions (e.g. `numpy.ma.zeros_like`,
```

**Review by rossbar — APPROVED** [review]
Seems like an improvement to me, thanks @mdhaber . A couple minor mispellings in the `ma` notes but other than that LGTM!

**Comment by mdhaber** [comment]
Oops, thanks @rossbar. Committed those changes. Feel free to squash merge.

**Pull request merged**